core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -3549,7 +3549,17 @@ protected final void createAggImpl(
     // implement HAVING (we have already checked that it is non-trivial)
     relBuilder.push(bb.root());
     if (havingExpr != null) {
-      relBuilder.filter(havingExpr);
+      /* Set the correlation variables used in this sub-query to the filter node,
+       * same logic is being used for the filter generated in where clause.
+       */
+      Set<CorrelationId> variableSet = new HashSet<>();
+      if (havingExpr instanceof RexSubQuery) {
```

Comment:
Agree that this might be taking care of limited cases, I didn't want to increase the scope of this PR. Please let me know if you foresee any use cases that is missing then I can take care of it in this PR (otherwise I think it might be better to handle it in another PR).

Reply:
Take an example: `select count(*) from emp group by ename having count(*) > 1 and (your correlated sub-query)`. This would not match this condition. We can easily handle this using `RexUtil.SubQueryFinder.find`, what do you think?

Reply:
Added the code changes to find the SubQuery and also a test case to exercise this code path. thanks